### PR TITLE
README fixes, refactored Configuration, bug fixes, easier generation of expected output files

### PR DESCRIPTION
**Pull request by jammua**
- Fixes to README.md
- Refactored and added Javadoc to Configuration
- Some bug fixes in Configuration and OracleGenerator
- Print line numbers during condition translator output to assist in creating expected output files
- Some other fixes here and there

### CODE REVIEW

File: src/main/java/org/toradocu/generator/OracleGenerator.java [hunk 1]
```diff
@@ -44,11 +45,14 @@ public static void createAspects(List<DocumentedMethod> methods) {
...
     String aspectDir = Toradocu.configuration.getAspectsOutputDir();
-    try {
-      Files.createDirectory(Paths.get(aspectDir));
-    } catch (IOException e) {
-      log.error("Error while creating" + aspectDir + ".", e);
-      System.exit(1);
+    Path aspectDirPath = Paths.get(aspectDir);
+    if (!Files.exists(aspectDirPath)) {
+      try {
+        Files.createDirectory(aspectDirPath);
```
Comment (albertogoffi) [inline src/main/java/org/toradocu/generator/OracleGenerator.java]: I think it's be better to keep the check on the existence of the aspectDirPath and the creation as an atomic instruction. I would rewrite the code as
```java
Path aspectDirPath = Paths.get(aspectDir);
try {
  Files.createDirectory(aspectDirPath);
} catch (FileAlreadyExistsException e) {
} catch (IOException e) {
  log.error("Error while creating directory \"" + aspectDir + "\".", e);
  System.exit(1);
}
```

Do you agree?

File: src/main/java/org/toradocu/conf/Configuration.java [hunk 2]
```diff
@@ -146,18 +138,18 @@
...
-  // Javadoc options
-
-  @DynamicParameter(names = "-J", description = "Javadoc options")
-  private Map<String, String> javadocOptions = new HashMap<>();
+  /** File used as template for generated aspects. */
+  private static final String ASPECT_TEMPLATE = "AspectTemplate.java";
 
-  private List<String> javadocOptionsList;
+  /** Command-line options passed to the Javadoc tool. */
+  private List<String> javadocOptions = new ArrayList<>();;
```
Comment (albertogoffi) [inline src/main/java/org/toradocu/conf/Configuration.java]: Fix the duplicated semicolon at the end of the line.

File: src/main/java/org/toradocu/conf/Configuration.java [hunk 3]
```diff
@@ -146,18 +138,18 @@
...
     names = "--aspects-output-dir",
-    description = "Directory where Toradocu will output aspects"
+    description = "Specifies a directory where Toradocu will output aspects"
   )
   private String aspectsOutputDir = "aspects";
 
-  // Javadoc options
-
-  @DynamicParameter(names = "-J", description = "Javadoc options")
-  private Map<String, String> javadocOptions = new HashMap<>();
```
Comment (albertogoffi) [inline src/main/java/org/toradocu/conf/Configuration.java]: How can the user of Toradocu specify options for the Javadoc tool? Why do we need to remove that possibility? Note that this change does not affect much the interface of Toradocu. It simply removes the -J option to pass through options to the Javadoc tool.

File: src/main/java/org/toradocu/generator/OracleGenerator.java (hunk 1 above)
Comment (mernst) [inline src/main/java/org/toradocu/generator/OracleGenerator.java]: An empty catch block is almost always a mistake.

**Comment by jammua** [comment]
I've made the two fixes requested above to the OracleGenerator and the Javadoc options command-line parameter.

**Pull request merged**